File: parq-backend/src/main/java/com/ParQ/ParQ/entity/BoardPost.java

package com.ParQ.ParQ.entity;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

import org.hibernate.annotations.CreationTimestamp;

import jakarta.persistence.CascadeType;
import jakarta.persistence.CollectionTable;
import jakarta.persistence.Column;
import jakarta.persistence.ElementCollection;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

@Entity
@Getter
@Setter
@Table(name = "boardPost")
public class BoardPost {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String title;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String content;

    @CreationTimestamp
    private LocalDateTime createdAt;

    @ManyToOne
    @JoinColumn(name = "userId", nullable = false)
    private User author;

    @OneToMany(mappedBy = "post", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    private List<BoardReply> replies = new ArrayList<>();

    @ElementCollection
    @CollectionTable(name = "boardPostFileUrls", joinColumns = @JoinColumn(name = "boardPostId"))
  
    private List<String> fileUrls = new ArrayList<>(); } 

File: ch_19_map/src/main/java/com/example/ch_19_map/BoardPost.kt
package com.example.ch_19_map

import com.google.gson.annotations.SerializedName
import java.io.Serializable
import java.time.LocalDateTime

data class BoardPost(
    @SerializedName("id") val id: Long,
    @SerializedName("title") val title: String,
    @SerializedName("content") val content: String,
    @SerializedName("createdAt") val createdAt: String, // JSON에서는 보통 String으로 받습니다.
    @SerializedName("author") val author: Author,
    @SerializedName("hasReply") val hasReply: Boolean, // 답변 존재 여부
    @SerializedName("fileUrls") val fileUrls: List<String>? = null // 이미지 경로 리스트 추가
) : Serializable

data class Author(
    @SerializedName("id") val id: Long,
    @SerializedName("username") val username: String,
    @SerializedName("email") val email: String
) : Serializable 